typescript peer dependency to low for successful compilation

### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 2.1.6
- __Running TSLint via__: Node.js API

TSLint cannot be used programmatically in a typescript project with `typescript<2.3.0` despite peerDependencies listing `typescript >= 2.1.0` i.e. writing a rule for TSLint with `typescript@~2.1.0` results in a compilation error:
 ```bash
node_modules/tslint/lib/language/utils.d.ts(52,57): error TS1005: ',' expected.
node_modules/tslint/lib/language/utils.d.ts(52,61): error TS1005: ',' expected.
```

Cause of this is https://github.com/palantir/tslint/blob/9b7574bce3b912d9ec71c9cf374b7d0423b0bb80/src/language/utils.ts#L106
Generic argument defaults where added in [typescript 2.3](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-3.html#generic-parameter-defaults).

Unfortunately this method is part of the public API which might cause trouble if we remove the default. Removing this default results in a successful build with 2.1.

Repro: [eps1lon/tslint-issue-typescript-peer-version](https://github.com/eps1lon/tslint-issue-typescript-peer-version)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,32 @@
 # Change Log
 
+## v5.18.0
+
+- [feature] New `--print-config` CLI flag (#4744)
+- [bugfix] [`no-unnecessary-type-assertion`](https://palantir.github.io/tslint/rules/no-unnecessary-type-assertion/) no longer flags `x!` non-null assertions if `--strictNullChecks` is not enabled (#3724)
+- [bugfix] [`quotemark`](https://palantir.github.io/tslint/rules/quotemark/) handles unbalanced quotes more gracefully without crashing (#4344)
+- [docs] Add tslint-playground link to README and bug report template (#4150)
+- [enhancement] [`no-magic-numbers`](https://palantir.github.io/tslint/rules/no-magic-numbers/) rule ignores arguments passed into default Number methods (#3668)
+- [new-fixer] [`comment-format`](https://palantir.github.io/tslint/rules/comment-format/) (#3583)
+- [new-rule-option] `"allow-trailing-lowercase"` option for [`comment-format`](https://palantir.github.io/tslint/rules/comment-format/) rule (#3583)
+- [new-rule-option] `"ignore-accessors"` for [`adjacent-overload-signatures`](https://palantir.github.io/tslint/rules/adjacent-overload-signatures/) to ignore getters / setters overloads (#3718)
+- [new-rule] [`no-async-without-await`](https://palantir.github.io/tslint/rules/no-async-without-await/) (#3945)
+- [new-rule] [`strict-comparisons`](https://palantir.github.io/tslint/rules/strict-comparisons/) (#4519)
+
+Thanks to our contributors!
+
+- Josh Goldberg
+- smoehrle
+- Mateusz Witkowski
+- Andy Hanson
+- John Wiseheart
+- Eran Shabi
+- Tibor Blenessy
+- AndreasGassmann
+- Ryan Waskiewicz
+- Isabelle
+
+
 ## v5.17.0
 
 - [bugfix] [`quotemark`](https://palantir.github.io/tslint/rules/quotemark/) backtic option now ignores enum members, use strict declarations, lookup types, and strings containing octal escape sequences. (#4693)

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
     "name": "tslint",
-    "version": "5.17.0",
+    "version": "5.18.0",
     "description": "An extensible static analysis linter for the TypeScript language",
     "bin": {
         "tslint": "./bin/tslint"
@@ -44,7 +44,7 @@
         "tsutils": "^2.29.0"
     },
     "peerDependencies": {
-        "typescript": ">=2.1.0 || >=2.1.0-dev || >=2.2.0-dev || >=2.3.0-dev || >=2.4.0-dev || >=2.5.0-dev || >=2.6.0-dev || >=2.7.0-dev || >=2.8.0-dev || >=2.9.0-dev || >=3.0.0-dev || >= 3.1.0-dev || >= 3.2.0-dev"
+        "typescript": ">=2.3.0-dev || >=2.4.0-dev || >=2.5.0-dev || >=2.6.0-dev || >=2.7.0-dev || >=2.8.0-dev || >=2.9.0-dev || >=3.0.0-dev || >= 3.1.0-dev || >= 3.2.0-dev"
     },
     "devDependencies": {
         "@octokit/rest": "^16.24.3",

--- a/src/linter.ts
+++ b/src/linter.ts
@@ -42,7 +42,7 @@ import { arrayify, dedent, flatMap, mapDefined } from "./utils";
  * Linter that can lint multiple files in consecutive runs.
  */
 export class Linter {
-    public static VERSION = "5.17.0";
+    public static VERSION = "5.18.0";
 
     public static findConfiguration = findConfiguration;
     public static findConfigurationPath = findConfigurationPath;

